app/util/notifications/hooks/index.test.ts

```diff
@@ -0,0 +1,249 @@
+
+    expect(mockNavigation.navigate).toHaveBeenCalledWith(
+      Routes.NOTIFICATIONS.DETAILS,
+      {
+        notificationId: notification.id,
+      },
+    );
+  });
+
+  it('should handle notifications correctly', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.PRESS,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  });
+
+  it('should do nothing if the EventType is DISMISSED', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DISMISSED,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+    });
+  });
+
+  it('should do nothing if data.action is not tx', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DELIVERED,
+          detail: {
+            notification: {
+              body: 'notificationTest',
+              data: {
+                action: 'no-tx',
+                id: '123',
+              },
+            },
+          },
+        }),
+      );
+
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+
+      expect(mockNavigate).not.toHaveBeenCalled();
+    });
+  });
+
+  it('handleOpenedNotification should do nothing if notification is null', async () => {
+    const { waitFor } = renderHook(() =>
+      useNotificationHandler(mockNavigation),
+    );
+
+    await act(async () => {
+      notifee.onForegroundEvent(() =>
+        mockNotificationEvent({
+          type: EventType.DELIVERED,
+          detail: {
+            notification: undefined,
+          },
+        }),
+      );
+      await waitFor(() => {
+        expect(notifee.onForegroundEvent).toHaveBeenCalled();
+      });
+
+      expect(mockNavigate).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should navigate to the transaction view when the notification action is "tx"', async () => {
```

Comment:
```suggestion
  it('navigates to the transaction view when the notification action is "tx"', async () => {
```
C: you are not testing what you say in the name, navigate call is not spied on.